BrowserKit/Sources/ToolbarKit/ToolbarButton.swift

```diff
@@ -74,7 +75,27 @@ class ToolbarButton: UIButton, ThemeApplicable {
         configuration = updatedConfiguration
     }
 
-    // MARK: ThemeApplicable
+    private func configureLongPressGestureRecognizerIfNeeded(for element: ToolbarElement) {
+        if element.hasLongPressGestureRecognizer {
+            let longPressRecognizer = UILongPressGestureRecognizer(
+                target: self,
+                action: #selector(handleLongPress)
+            )
+            addGestureRecognizer(longPressRecognizer)
+        }
+    }
+
+    // MARK: - Selectors
+    @objc
+    private func handleLongPress(_ gestureRecognizer: UILongPressGestureRecognizer) {
+        if gestureRecognizer.state == .began {
+            let generator = UIImpactFeedbackGenerator(style: .light)
+            generator.impactOccurred()
+            // Handle the long press
```

Comment:
The `ToolbarElement` needs a closure for the long press action that would be called here.